Topics_2_3/src/main/java/ShoppingCart/UserServiceImp.java

```diff
@@ -0,0 +1,89 @@
+package ShoppingCart;
+
+/**
+ * This class contains the implementation for UserService CRUD operations.
+ * 
+ * @author Victor Munareto
+ *
+ */
+public class UserServiceImp implements UserService {
+
+	Database databaseConnection;
+
+	/**
+	 * The constructor creates an instance of the class and connects it with the
+	 * database.
+	 * 
+	 * @param database
+	 */
+	public UserServiceImp(Database database) {
+		this.databaseConnection = database;
+	}
+
+	/**
+	 * This method registers a new user to the database with the given information.
+	 * 
+	 * @param name
+	 * @param password
+	 * @param document
+	 * @param telephone
+	 * @param email
+	 */
+	@Override
+	public String registerUser(String name, String password, String document, String telephone, String email) {
+
+		databaseConnection.userDatabase.add(new User(name, password, document, telephone, email));
```

Comment:
To properly make UserServiceImp independent of the Database would be better to *not* access directly to Database field (userDatabase).